### ISSUE TITLE
impletmet complex branch

### ISSUE THREAD

**Issue by kev7blu** [ece-231l-fall-2019/week-4-complex-skyler-s-team#2]

**Comment by lilacnight** [comment]
If you go to mediasite lectures David talks about complex math in out last thursday lecture at about time 1:14:30.

**Comment by kev7blu** [comment]
I feel like we are missing some function based on what I saw from class yesterday like
"+= operator" and "-=operator".....

**Comment by kev7blu** [comment]
also work on tests for the other functions

**Comment by kev7blu** [comment]
I am currently debugging the dividing and implementing stuff to it

**Comment by lilacnight** [comment]
Alright sorry I didn't see this sooner what functions do you have ready to be tested?

**Comment by kev7blu** [comment]
all the functions except the += etc operators

**Comment by lilacnight** [comment]
alright did you see the function implementation for the = operator? The Complex& functions inside the complex class need to be implemented a certain way like this one is

**Comment by kev7blu** [comment]
yeah I just trying to figure out the right way

**Comment by lilacnight** [comment]
The norm operator is having some issues, almost all the other new operators are passing

**Comment by lilacnight** [comment]
Yeah everything is working well besides norm and conjugate.

**Comment by kev7blu** [comment]
ok I'll have a look

**Comment by kev7blu** [comment]
are you sure you have the right answers in assertion?

b/c I have stared at my computer and looked for bugs and only found one

FYI 
norm -  returns the value of a squared magnitude which will return a complex number in the form
----> a (+or-) bi

conjuncate - returns a real value
for it should not have a value of i inside

**Comment by lilacnight** [comment]
I thought conjugate was just a+bi --> a-bi

**Comment by lilacnight** [comment]
Sorry I had to get dinner for my fam. I have the tests for the += operators and stuff ready.

**Comment by lilacnight** [comment]
I believe norm should return a real number since it is the magnitude of the square and the complex conjugate should just times the imaginary element by -1. 

double norm(const Complex& z);
Complex conj(const Complex& z);

Here you can see the norm returns only a real value of type double and the conjugate returns a complex number.

**Comment by kev7blu** [comment]
do a problem on paper and you'll see what i mean:

(2 + 2i)^2

and 

(2+ 2i)(2-2i)

**Comment by lilacnight** [comment]
The square magnitude would be 8 since it's just the real and imaginary parts squared and added together - the norm function should return a double of value 8 in this case. Here's an article explaining a complex conjugate. https://en.wikipedia.org/wiki/Complex_conjugate 
For 2+2i it would just be 2-2i, there is no need for any calculations.

**Comment by kev7blu** [comment]
wait are you serious?
so all that is needed is the conjugate wow thats alittle too easy

for norm thats correct it should return a real

**Comment by lilacnight** [comment]
Yeah that's all. Okay so we're are on the same page with the norm cool. Do you have any other functions ready for testing?

**Comment by kev7blu** [comment]
just the += -= and *= thats it

you merge it IDK how to do so ...

**Comment by lilacnight** [comment]
Okay I'm going to merge your branch into mine and test what you have so far then I'll pass back the runtests to your branch so you can see what failed or passed

**Comment by kev7blu** [comment]
You do it?

**Comment by lilacnight** [comment]
Yeah I created a pull request go review the differences real quick then accept the request (keep the version of conj from my file) and run the test

**Comment by lilacnight** [comment]
I don't think we're gonna get this done by tonight wanna just save the rest for monday lab?

**Comment by kev7blu** [comment]
I finally understand this but it's just we need to just check with David to see we did everything right

**Comment by kev7blu** [comment]
the norm is the squared magnitude so the that's just      sqrt((-20)^2 + (-8i)^2 )?

**Comment by kev7blu** [comment]
sure.

**Comment by lilacnight** [comment]
yeah that's right. Alright cool, I don't think we'll have much left

**Comment by kev7blu** [comment]
did you fix the changes that were causing problems?

**Comment by kev7blu** [comment]
b/c this is due @ 5pm
